### PR TITLE
Scrum_4_post_presentations

### PR DESCRIPTION
**Pull request by Jasonn5**
This Pull Request introduces a series of improvements and additions to various modules within the project, focusing on API request management, schema validation, and unit tests for meeting endpoints. Below are the details of the changes made:

Changes in api_request.py
POST Methods:

- Improved the existing post method.
- Added post_json and post_without_headers methods for greater flexibility in POST requests.
- Added the delete method to remove resources.

Specific Changes:

- post method: maintains its original functionality.
- post_json method: allows sending payloads in JSON format.
- post_without_headers method: enables making POST requests without headers.
- delete method: allows making DELETE requests.
- Changes in meeting_endpoints.py

New Methods:

- Added get_meeting_with_params method to fetch meetings with specific parameters.
- Added delete_meeting method to delete meetings.
- Changes in assertion_schemas.py

New Assertion Functions:

- assert_payload_presentation_schema: added to validate the presentation of the meeting payload according to the defined JSON schema.

New Schema:

- payload_meeting_schema.json: defines the structure and data types expected for the meeting payload.
- Changes in confest.py

Pytest Fixtures:

- Added fixtures to handle the base payload loading and cleanup of meeting data post-test.
- Tests in test_get_presentations.py and test_post_meeting.py

New Tests:

- Added tests to validate the operations of fetching and creating meetings, including:
- Validation of successful meeting retrieval.
- Response schema validation.
- Authentication and authorization tests.
- Pagination and sorting tests for meetings.
- Creation tests for meetings with various authentication scenarios and payloads.
- Objectives of the Pull Request
- Enhance the flexibility and robustness of API requests.
- Ensure data integrity through strict payload validations.
- Strengthen test coverage to guarantee the correct functioning of meeting endpoints under multiple scenarios.
- Review and Approval

We request a review of these changes to ensure that:

The new functionalities meet the specified requirements.

- The schema validations are correct and comprehensive.
- The tests adequately cover the intended use cases and ensure system stability.

### CODE REVIEW

File: api_endpoints/api_request.py
Comment (Diego-Claros17) [inline api_endpoints/api_request.py:1]: GJ

File: api_endpoints/meeting_endpoints.py
Comment (Diego-Claros17) [inline api_endpoints/meeting_endpoints.py:1]: No duplicated files

File: assertions/assertion_schemas.py
Comment (Diego-Claros17) [inline assertions/assertion_schemas.py:1]: Well done

File: resources/schemas/payload_meeting_schema.json
Comment (Diego-Claros17) [inline resources/schemas/payload_meeting_schema.json:1]: Aligned with the new structure

File: tests/presentations/conftest.py
Comment (Diego-Claros17) [inline tests/presentations/conftest.py:1]: GJ

File: tests/presentations/test_get_presentations.py
Comment (Diego-Claros17) [inline tests/presentations/test_get_presentations.py:1]: Well done

**Review by Diego-Claros17 — APPROVED** [review]
No problems and no conflicts, good job Jeyson,

File: api_endpoints/api_request.py
Comment (AlisonGuzman2006) [inline api_endpoints/api_request.py:1]: The change has been reviewed and approved

File: api_endpoints/meeting_endpoints.py
Comment (AlisonGuzman2006) [inline api_endpoints/meeting_endpoints.py:1]: The change has been reviewed and approved

File: assertions/assertion_schemas.py
Comment (AlisonGuzman2006) [inline assertions/assertion_schemas.py:1]: The change has been reviewed and approved

File: payloads/meeting/payload_meeting.json
Comment (AlisonGuzman2006) [inline payloads/meeting/payload_meeting.json:1]: The addition has been reviewed and approved

File: resources/schemas/payload_meeting_schema.json
Comment (AlisonGuzman2006) [inline resources/schemas/payload_meeting_schema.json:1]: The addition has been reviewed and approved

File: tests/presentations/conftest.py
Comment (AlisonGuzman2006) [inline tests/presentations/conftest.py:1]: The addition has been reviewed and approved

File: tests/presentations/test_get_presentations.py
Comment (AlisonGuzman2006) [inline tests/presentations/test_get_presentations.py:1]: The change has been reviewed and approved

File: tests/presentations/test_post_meeting.py
Comment (AlisonGuzman2006) [inline tests/presentations/test_post_meeting.py:1]: The addition has been reviewed and approved

**Review by AlisonGuzman2006 — APPROVED** [review]
The changes, and addition has been reviewed and approved

**Pull request merged**